discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -1,73 +1,79 @@
+import logging
 import libvirt
-import utils
+from test_infra import consts
+from test_infra.controllers.node_controllers.node_controller import NodeController
 
-import consts
-from node_controlers.node_controler import NodeControler
 
+class LibvirtController(NodeController):
 
-class QeVmController(NodeControler):
-
-    def __init__(self):
+    def __init__(self, **kwargs):
         self.libvirt_connection = libvirt.open('qemu:///system')
 
     def __del__(self):
         self.libvirt_connection.close()
 
     def list_nodes(self):
+        return self.list_nodes_with_name_filter(None)
+
+    def list_nodes_with_name_filter(self, name_filter):
+        logging.info("Listing current hosts with name filter %s", name_filter)
         nodes = {}
         domains = self.libvirt_connection.listAllDomains()
-
         for domain in domains:
             domain_name = domain.name()
+            if name_filter and name_filter not in domain_name:
```

Comment:
```python
 if not name_filter or name_filter not in domain_name
```
Makes more sense

Reply:
how? don't agree

Reply:
Because you want to filter only if you have `name_filter`, if not - continue
If you have `name_filter` -> check whether it's not in the `domain_name`